pkg/storaged/block/format-dialog.jsx

```diff
@@ -532,11 +552,19 @@
                                                    })
                                 .then(() => maybe_unlock())
                                 .then(content_block => {
-                                    return content_block.Format(vals.type, options);
+                                    return content_block.Format(type, options);
                                 }));
                     } else {
-                        return block.Format(vals.type, options);
+                        return block.Format(type, options);
+                    }
+                }
+
+                async function set_partition_type(new_path) {
+                    if (partition_type) {
+                        const block_part = await client.wait_for(() => client.blocks_part[new_path]);
+                        await block_part.SetType(partition_type, {});
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240109-133522-cb1438ac-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)